android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStoragePackage.java

```diff
@@ -29,6 +31,7 @@
                 Class storageClass = Class.forName("com.reactnativecommunity.asyncstorage.next.StorageModule");
                 NativeModule inst = (NativeModule) storageClass.getDeclaredConstructor(new Class[]{ReactContext.class}).newInstance(reactContext);
                 moduleList.add(inst);
+                AsyncLocalStorageUtil.verifyAndForceSqliteCheckpoint(reactContext);
```

Comment:
Shouldn't this line be added wherever we're doing the migration? Won't this force a checkpoint on every startup?

Reply:
Yes, doing this here will basically check to do the checkpoint on every app startup. Ideally, we should do it only when the migration will be performed, but from what I checked, the migration process is fired after the new db file is created - meaning that forcing the checkpoint no longer make sense, since the old db file was already read.

I think it's fine to leave it here - we make sure we fire checkpoint only once, by checking if existence states of both db files.